### ISSUE TITLE
Incorrect position of width points on outline when opened old files in 1.3.13

### ISSUE THREAD

**Issue by morevnaproject** [synfig/synfig#1284]
**Synfig version & platform**:
Synfig 1.3.12, 1.3.13

**Issue description**:

[outline-test-2.sif.zip](https://github.com/synfig/synfig/files/4380047/outline-test-2.sif.zip)

In Synfig 1.0.2 / 1.2.2 / 1.3.11:
![screenshot_008](https://user-images.githubusercontent.com/332868/77522396-c798bb00-6eb6-11ea-98b2-af15d601015a.png)

In Synfig 1.3.12, 1.3.13:
![screenshot_009](https://user-images.githubusercontent.com/332868/77522442-d7b09a80-6eb6-11ea-85ec-a12af250afbc.png)

Detailed description of this issue: https://github.com/synfig/synfig/issues/314#issuecomment-603703084

How to solve:
Modify file loading function. It should check if file is old (check canvas version) and if it is, then for all looped outlines change order of vertices (take last vertex and make it first).

**Comment by morevnaproject** [comment]
Good news:
- Synfig 1.3.11 uses canvas version 1.0
- Synfig 1.3.12 uses canvas version 1.1

https://github.com/synfig/synfig/blob/1db7f4d4952355dd844a6e0fb4924ecda2f3df3c/synfig-core/src/synfig/canvas.h#L117-L131

**Comment by morevnaproject** [comment]
@rodolforg Thank you for the fix! ^__^

**Comment by morevnaproject** [comment]
I have noticed this condition in code:
```
if (gran_parent && gran_parent->get_name() == "layer" && gran_parent->get_attribute_value("type") == "advanced_outline") {
```
which makes the fix applied to Advanced outlines only.

I think we should apply this fix to ALL blines - including Regular Outlines, Regions, even if they are not belong to layer (exported).
This will fix issue demonstrated in https://github.com/synfig/synfig/issues/314#issuecomment-601336943

We need to apply fix to all blines, because any bline can report position, in case if it is referenced in Spline Vertex convert (like in case of https://github.com/synfig/synfig/issues/314#issuecomment-601336943 - the advanced outline is attached to Regular Outline, which reports wrong poisions and so the artwork gets distorted)

**Comment by morevnaproject** [comment]
P.S. Of course, I was talking about all _looped_ blines. For unlooped we
shouldn’t change anything. ^__^
-- 
http://morevnaproject.org/

**Comment by morevnaproject** [comment]
> I think we should apply this fix to ALL blines - including Regular Outlines, Regions, even if they are not belong to layer (exported).

Fixed with a714f50d64b97f58eed8b01.

P.S. Apologies, I accidentally committed directly to master branch, instead of creating PR. Will be more careful next time. ^__^"

**Comment by rodolforg** [comment]
@morevnaproject  So, I should use canvas version 1.0 to detect the conversion need.
Is Version >= 1.1 fine?
Is there any version previous to 1.0?

**Comment by rodolforg** [comment]
@morevnaproject  this solution of yours doesn't check for any version. Am I right?

I propose this fix:
```cpp
			if (loop) {
				string version = canvas->get_version();
				if (version == "1.0" || (version[0] == '0' && version[1] == '.'))
					must_rotate_point_list = true;
			}
```

**Comment by morevnaproject** [comment]
Ah, good catch! At first I didn't understood what's the difference in "checking version of outline layer" vs "check version of canvas". I actually got the impression that you found a better way than I suggested in https://github.com/synfig/synfig/issues/1284#issuecomment-603847532 ^__^

But then I found that  Region Layer doesn't has same version as Outline Layer. So we have to check canvas version, you right. ^__^
